### DISCUSSION TITLE
Shell Completion through postInstall Hook

### DISCUSSION THREAD

**Discussion by sinrohit** [ipetkov/crane#668] [Q&A] [answered]
```
postInstall = ''
            installShellCompletion --cmd <name> \
              --bash <($out/bin/<name> completion bash) \
              --zsh <($out/bin/<name> completion zsh) \
              --fish <($out/bin/<name> completion fish)
          '';
```

This looks for bin in `<name>-deps-<version>` outpath  instead of looking in ` <name>-<version>` outpath.

**Comment by ipetkov** [comment, marked as answer]
Hi @rsrohitsingh682, normally when you call `buildPackage` _without_ specifying `cargoArtifacts` all attributes will be applied to the deps-only and the final derivations. Since you want the install hook to only run on the final derivation, you'll need to explicitly call `buildDepsOnly` with the install hook omitted:

```nix
let
  commonArgs = {
    # everything except for `postInstall`
  };
  cargoArtifacts = craneLib.buildDepsOnly commonArgs;
in
craneLib.buildPackage (commonArgs // {
  inherit cargoArtifacts;
  postInstal = ''
    # etc ...
  '';
})
```
